src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -784,35 +781,34 @@ private BoundPattern BindRecursivePattern(RecursivePatternSyntax node, TypeSymbo
             }
 
             // Ensure ITuple has a Length and indexer
-            iTupleGetLength = (MethodSymbol)Compilation.GetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_ITuple__get_Length, recordUsage: false);
-            iTupleGetItem = (MethodSymbol)Compilation.GetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_ITuple__get_Item, recordUsage: false);
+            iTupleGetLength = (MethodSymbol)Compilation.GetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_ITuple__get_Length);
+            iTupleGetItem = (MethodSymbol)Compilation.GetWellKnownTypeMember(WellKnownMember.System_Runtime_CompilerServices_ITuple__get_Item);
             if (iTupleGetLength is null || iTupleGetItem is null)
             {
                 // This might not result in an ideal diagnostic
                 return false;
             }
 
             // passed all the filters; permit using ITuple
-            if (!IsSemanticModelBinder)
-            {
-                AddUsedAssembly(iTupleType.ContainingAssembly);
-            }
+            _ = diagnostics.ReportUseSite(iTupleType, node) ||
+                diagnostics.ReportUseSite(iTupleGetLength, node) ||
+                diagnostics.ReportUseSite(iTupleGetItem, node);
```

Comment:
Why bother with these other two? Is there a scenario where this will result in missing use sites?

Reply:
And if there is, please add a comment, because it's very much not obvious.

---
In reply to: [361552846](https://github.com/dotnet/roslyn/pull/40560#discussion_r361552846) [](ancestors = 361552846)

Reply:
> Why bother with these other two? Is there a scenario where this will result in missing use sites?

> And if there is, please add a comment, because it's very much not obvious.

One should not assume that use site information for one member is the same as for the other. This was true before my change and remains true with it. The general rule is - you use a symbol, you report use site for it. Following this rule doesn't need any comment saying that the rule is being followed.


---
In reply to: [361552882](https://github.com/dotnet/roslyn/pull/40560#discussion_r361552882) [](ancestors = 361552882,361552846)

Reply:
> One should not assume that use site information for one member is the same as for the other.

Right, which is why I'm very confused here. If we successfully reported the use site for the first one, we won't report the use site of the other two as the `||` will short circuit. So I do not understand what you are trying to do here.

---
In reply to: [361557934](https://github.com/dotnet/roslyn/pull/40560#discussion_r361557934) [](ancestors = 361557934,361552882,361552846)

Reply:
> Right, which is why I'm very confused here. If we successfully reported the use site for the first one, we won't report the use site of the other two as the || will short circuit. So I do not understand what you are trying to do here.

The ```ReportUseSite``` API returns true if an error was reported that is how all APIs around use site reporting used to work and keep working. It is obvious that one error is good enough to report (we are trying to avoid as much noise as possible in the same manner all over the place). We also do not care about tracking dependencies when we have an error. That is also a common theme across the entire feature.  

---
In reply to: [361710019](https://github.com/dotnet/roslyn/pull/40560#discussion_r361710019) [](ancestors = 361710019,361557934,361552882,361552846)